zuul-core/src/main/java/com/netflix/zuul/netty/filter/BaseZuulFilterRunner.java

```diff
@@ -255,22 +255,23 @@ protected void recordFilterCompletion(final ExecutionStatus status, final ZuulFi
                                           final ZuulMessage zuulMesg, final ZuulMessage startSnapshot) {
 
         final SessionContext zuulCtx = zuulMesg.getContext();
-        final long execTime = System.currentTimeMillis() - startTime;
-        if (execTime >= FILTER_EXCESSIVE_EXEC_TIME.get()) {
-            LOG.warn("Filter {} took {} ms to complete! status = {}", filter.filterName(), execTime, status.name());
+        final long execTimeNs = System.nanoTime() - startTime;
+        final long execTimeMs = execTimeNs / 1_000_000L;
+        if (execTimeMs >= FILTER_EXCESSIVE_EXEC_TIME.get()) {
+            LOG.warn("Filter {} took {} ms to complete! status = {}", filter.filterName(), execTimeMs, status.name());
```

Comment:
One other meta comment: this should be a counter right?   It may be difficult to aggregate across logs to find slow filters.